### DISCUSSION TITLE
get client IP without lose SSR

### DISCUSSION THREAD

**Discussion by EverStarck** [vercel/next.js#22098] [Help]
Hi. 
 I'm learning and making a wheater app, so, I need the user IP. I used IPIFY but this returned the server API or I lose the SSR.
  So, I see that I can get the client IP from the HTTP headers, and I made an API with nextjs routes, but I need to call it with `http://localhost:3000/api`.
   Then I see this [issue](https://github.com/vercel/next.js/issues/9524#issuecomment-589772756), and I implement it but I don't know how to pass the req to the getStaticProps on  index.js
   
   **/api/index**

    export async function getData(passReq) {
	  let myData =  passReq.headers["x-forwarded-for"] || passReq.connection.remoteAddress;
	  
      //remove ::ffff:
      if (myData.substr(0, 7) == "::ffff:") {
        myData = myData.substr(7);
      } 
      return myData;
    } 

    export default async (req, res) => {
       const myData = await getData(req);
       res.json({ myData });
    };

**index pages**

    import {getData} from './api/index';
    
    export default function Home({ myData }) {
	   console.log(myData) //this returns undefined
	   return(...)
	}
	
	export async function getStaticProps() { 
	   const myData = await getData(req)

	   return { props: { myData } };
	}

  
   Any idea or other solution?

**Comment by mkrtchian** [comment]
Unfortunately `getStaticProps` is executed only on the server, and only at build time. So I think it can't know information related to a specific request (which happens after the build).

You can use `getServerSideProps` instead. Your page will be slower but you will be able to execute code on your server for a specific request like what you want to do :

```javascript
export async function getServerSideProps(context) {
  const myData = await getData(context.req);

  return { props: { myData } };
}
```
